rasgotransforms/rasgotransforms/render.py

```diff
@@ -0,0 +1,93 @@
+"""
+Functions to render Transform Templates
+"""
+from typing import Dict, Any, Optional
+from pathlib import Path
+from .main import DataWarehouse
+from .dtypes import DTYPES
+from importlib import machinery, util
+import os
+import re
+
+
+def get_dtype(dtype: str) -> str:
+    return DTYPES[dtype.lower()]
+
+
+def get_dw_type(dw_type: str) -> Optional[DataWarehouse]:
+    if dw_type:
+        try:
+            return DataWarehouse[dw_type.upper()]
+        except KeyError:
+            raise Exception(f'Unsupported DataWarehouse type: {dw_type}')
+    else:
+        return None
+
+
+def get_path(transform_name: str, dw_type: Optional[DataWarehouse] = None) -> Optional[str]:
+    root_dir = os.path.dirname(__file__)
+    if dw_type:
+        function_path = Path(root_dir, 'transforms', transform_name, dw_type.value,
+                             f'{transform_name}.py')
+        if os.path.exists(function_path):
+            return str(function_path.absolute())
+    function_path = Path(root_dir, 'transforms', transform_name, f'{transform_name}.py')
+    if not os.path.exists(function_path):
+        return None
+    return str(function_path.absolute())
+
+
+def import_from(module, name):
+    module = __import__(module, fromlist=[name])
+    return getattr(module, name)
+
+
+def cleanse_name(symbol: str) -> str:
+    """
+    Extra verbose function for clarity
+
+    remove double quotes
+    replace spaces and dashes with underscores
+    cast to upper case
+    delete anything that is not letters, numbers, or underscores
+    if first character is a number, add an underscore to the beginning
+    """
+    symbol = str(symbol)
+    symbol = symbol.strip()
+    symbol = symbol.replace(' ', '_').replace('-', '_')
+    symbol = symbol.upper()
+    symbol = re.sub('[^A-Z0-9_]+', '', symbol)
+    # if symbol is empty, at least returns '_'
+    # if starts with a decimal prefix with '_'
+    symbol = '_' + symbol if symbol[0].isdecimal() or not symbol else symbol
```

Comment:
```suggestion
    symbol = f'_{symbol}' if not symbol or symbol[0].isdecimal() else symbol
```

You want to do the truthy check first so it won't indexerror if there isn't an index 0. 

Also use fstrings or format, and [never + to concat strings](https://codecalamity.com/stop-using-plus-signs-to-concatenate-strings/)

Reply:
if they symbol is empty, do we want to error instead?